common/repo/GitRepo.java

```diff
@@ -123,7 +123,7 @@ private String formatError(CommandResult commandResult) {
 
   @Override
   public void switchBranch(String branch) {
-    if (branchExists(branch)) {
+    if (branchExists(branch) || commitExists(branch)) {
```

Comment:
Why are we passing a branch to `commitExists()`? Doesn't it expect a commit sha?

Reply:
@oferb _technically_ a 'branch' is just a fancy name for commit SHA, that is -
both `git cat-file -t master` and `git cat-file -t 104058642173e7a6e807047b0d19a909a4cfa6ff` output `commit`
(`git cat-file` is used by `GitRepo::commitExists`)

Reply:
Got it :)